internal/api/courses/assignments/submissions/proxy/resubmit.go

```diff
@@ -0,0 +1,113 @@
+package proxy
+
+import (
+	"github.com/edulinq/autograder/internal/api/core"
+	"github.com/edulinq/autograder/internal/db"
+	"github.com/edulinq/autograder/internal/grader"
+	"github.com/edulinq/autograder/internal/log"
+	"github.com/edulinq/autograder/internal/model"
+	"github.com/edulinq/autograder/internal/timestamp"
+	"github.com/edulinq/autograder/internal/util"
+)
+
+type ResubmitRequest struct {
+	core.APIRequestAssignmentContext
+	core.MinCourseRoleGrader
+	TargetSubmission string `json:"target-submission"`
+
+	ProxyUser core.TargetCourseUser `json:"proxy-email"`
+	ProxyTime *timestamp.Timestamp  `json:"proxy-time"`
+
+	Message string `json:"message"`
+}
+
+type ResubmitResponse struct {
+	FoundUser       bool   `json:"found-user"`
+	FoundSubmission bool   `json:"found-submission"`
+	Rejected        bool   `json:"rejected"`
+	Message         string `json:"message"`
+
+	GradingSuccess bool               `json:"grading-success"`
+	GradingInfo    *model.GradingInfo `json:"result"`
+}
+
+// Proxy resubmit an assignment submission to the autograder.
+func HandleResubmit(request *ResubmitRequest) (*ResubmitResponse, *core.APIError) {
+	response := ResubmitResponse{}
+
+	if !request.ProxyUser.Found {
+		return &response, nil
+	}
+
+	response.FoundUser = true
+
+	gradingResult, err := db.GetSubmissionContents(request.Assignment, request.ProxyUser.Email, request.TargetSubmission)
+	if err != nil {
+		return nil, core.NewInternalError("-632", &request.APIRequestCourseUserContext, "Failed to get submission contents.").
+			Err(err).Assignment(request.Assignment.GetID()).
+			Add("target-user", request.ProxyUser.Email).Add("submission", request.TargetSubmission)
+	}
+
+	if gradingResult == nil {
+		return &response, nil
+	}
+
+	response.FoundSubmission = true
+
+	tempDir, err := util.MkDirTemp("resumbit-request-files-")
+	if err != nil {
+		return nil, core.NewInternalError("-633", &request.APIRequestCourseUserContext, "Failed to create temp resubmit files directory.").
+			Err(err).Assignment(request.Assignment.GetID()).
+			Add("target-user", request.ProxyUser.Email).Add("submission", request.TargetSubmission)
+	}
+
+	err = util.GzipBytesToDirectory(tempDir, gradingResult.InputFilesGZip)
+	if err != nil {
+		util.RemoveDirent(tempDir)
```

Comment:
Instead of removing here, defer the remove on 63 (right after the error check for its creation).

(Of course, temp dirs get automatically removed, but we like to keep it clean (since the automatic removal is not until shutdown).)